A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/ATSU/A320_Neo_CDU_AOC_OFPData.js

```diff
@@ -0,0 +1,546 @@
+                CDUAocOfpData.ShowPage(mcdu);
+            }
+        }
+
+        let blockFuel = "_____[color]amber";
+        let taxiFuel = "____[color]amber";
+        let tripFuel = "_____[color]amber";
+        let requestButton = "SEND*[color]cyan";
+        let loadButton = "*LOAD[color]cyan";
+
+        if (mcdu.simbrief.sendStatus !== "READY" && mcdu.simbrief.sendStatus !== "DONE") {
+            requestButton = "SEND [color]cyan";
+        }
+
+        if (mcdu.aocWeight.loading) {
+            loadButton = " LOAD[color]cyan";
+        }
+
+        const currentBlockFuel = mcdu.aocWeight.blockFuel || mcdu.simbrief.blockFuel;
+        if (currentBlockFuel) {
+            const size = mcdu.aocWeight.blockFuel ? 'big' : 'small';
+            blockFuel = `{${size}}${currentBlockFuel}{end}[color]cyan`;
+        }
+
+        const currentTaxiFuel = mcdu.aocWeight.taxiFuel || mcdu.simbrief.taxiFuel;
+        if (currentTaxiFuel) {
+            const size = mcdu.aocWeight.taxiFuel ? 'big' : 'small';
+            taxiFuel = `{${size}}${currentTaxiFuel}{end}[color]cyan`;
+        }
+
+        const currentTripFuel = mcdu.aocWeight.tripFuel || mcdu.simbrief.tripFuel;
+        if (currentTripFuel) {
+            const size = mcdu.aocWeight.tripFuel ? 'big' : 'small';
+            tripFuel = `{${size}}${currentTripFuel}{end}[color]cyan`;
+        }
+
+        const display = [
+            ["PERF/W&B", "1", "2", "AOC"],
+            ["BLOCK FUEL"],
+            [blockFuel],
+            ["TAXI FUEL"],
+            [taxiFuel],
+            ["TRIP FUEL"],
+            [tripFuel],
+            [""],
+            ["", "PRINT*[color]inop"],
+            ["REFUEL", "OFP REQUEST[color]cyan"],
+            [loadButton, requestButton],
+            [""],
+            ["<AOC MENU"]
+        ];
+        mcdu.setTemplate(display);
+
+        mcdu.leftInputDelay[0] = () => {
+            return mcdu.getDelayBasic();
+        };
+        mcdu.onLeftInput[0] = (value) => {
+            if (value === FMCMainDisplay.clrValue) {
+                mcdu.aocWeight.blockFuel = "";
+                updateView();
+                return true;
+            }
+            const maxAllowableFuel = 21273;
+            const enteredFuel = Math.round(+value);
+            if (enteredFuel >= 0 && enteredFuel <= maxAllowableFuel) {
+                mcdu.aocWeight.blockFuel = enteredFuel.toString();
+                updateView();
+                return true;
+            }
+            mcdu.showErrorMessage(mcdu.defaultInputErrorMessage);
+            return false;
+        };
+
+        mcdu.leftInputDelay[1] = () => {
+            return mcdu.getDelayBasic();
+        };
+        mcdu.onLeftInput[1] = (value) => {
+            if (value === FMCMainDisplay.clrValue) {
+                mcdu.aocWeight.taxiFuel = "";
+                updateView();
+                return true;
+            }
+            const maxAllowableFuel = 21273;
```

Comment:
I would be very careful with this, does this match our documents? Are we able to read it from the sim directly because this is defined in the cfg file of the aircraft.